api/api/version_endpoints_api.go

```diff
@@ -53,7 +53,7 @@ func (c *EndpointsController) ListEndpoint(r *http.Request, vars map[string]stri
 
 	version, err := c.VersionsService.FindByID(ctx, modelID, versionID, c.MonitoringConfig)
 	if err != nil {
-		return NotFound(fmt.Sprintf("Version with given `version_id: %d` not found", versionID))
+		return NotFound(fmt.Sprintf("Version with given `version_id: %d` not found: %s", versionID, err))
```

Comment:
Shall we also log the error to the console for any successful return?

Reply:
Yes, sounds good. I'm thinking about logging from the utility methods for `NotFound` and `InternalServerError` [here](https://github.com/caraml-dev/merlin/blob/main/api/api/response.go#L97), and clean up the individual log statements from the API handlers. Wdyt?

Reply:
I think it's better to do it in the [router](https://github.com/caraml-dev/merlin/blob/75853498bb3cd3069553858f2332d4101119e9c8/api/api/router.go#LL121C4-L121C10). Although need to check whether the stack trace is still preserved at that point.

Reply:
It seems like previously you were consistently using `%v` to format the error, but it's `%s` in this case. Just wondering if it's intentional 🤔

Reply:
For default errors, they are equivalent. But I correct this to `%v` to be consistent. Thanks for catching!